### PR TITLE
Enhancement/user signup

### PR DESCRIPTION
**Pull request by SubhankarPradhan88**
For Signup endpoint below classes / objects has been created:
1. Signup Controller
2. UserEntity
3. Created Signup Business service
4. User Dao

### CODE REVIEW

File: quora-api/src/main/java/com/upgrad/quora/api/controller/SignupController.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/")
+public class SignupController {
```
Comment (mshashank0) [inline quora-api/src/main/java/com/upgrad/quora/api/controller/SignupController.java]: Rename/refector "SignupController" to "UserController". It is mentioned in "REST API endpoints - 1". Please follow the naming convention.

File: quora-service/src/main/java/com/upgrad/quora/service/business/SignupBusinessService.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.upgrad.quora.service.dao.UserDao;
+import com.upgrad.quora.service.entity.UserEntity;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class  SignupBusinessService {
```
Comment (mshashank0) [inline quora-service/src/main/java/com/upgrad/quora/service/business/SignupBusinessService.java]: Rename/refector "SignupBusinessService" to "UserBusinessService".

File: quora-api/src/main/java/com/upgrad/quora/api/controller/SignupController.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Autowired
+    private SignupBusinessService signupBusinessService;
+    @RequestMapping(method = RequestMethod.POST, path = "/signup", consumes = MediaType.APPLICATION_JSON_UTF8_VALUE, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
```
Comment (mshashank0) [inline quora-api/src/main/java/com/upgrad/quora/api/controller/SignupController.java]: Change the path = "/signup" to path = "/user/signup".  Please follow the guidelines for naming convention.

File: quora-api/src/main/java/com/upgrad/quora/api/controller/SignupController.java (hunk 1 above)
Comment (SubhankarPradhan88) [reply quora-api/src/main/java/com/upgrad/quora/api/controller/SignupController.java]: "SignupController" name changed to "UserController".

File: quora-api/src/main/java/com/upgrad/quora/api/controller/SignupController.java (hunk 3 above)
Comment (SubhankarPradhan88) [reply quora-api/src/main/java/com/upgrad/quora/api/controller/SignupController.java]: Path has been updated to "/user/signup".

**Comment by mshashank0** [comment]
Looks good to me. @post2shyam Please review and merge

**Pull request merged**